server/models.py

```diff
@@ -0,0 +1,581 @@
+
+    # The definition file contains election related data
+    definition_file_id = Column(
+        String(200), ForeignKey("file.id", ondelete="set null")
+    )
+    definition_file = relationship(
+        "File",
+        foreign_keys=[definition_file_id],
+        single_parent=True,
+        cascade="all, delete-orphan",
+    )
+
+    # Contests in the election
+    contests = relationship(
+        "Contest",
+        back_populates="election",
+        uselist=True,
+        passive_deletes=True,
+        order_by="Contest.name",
+    )
+
+    # Parties in the election
+    parties = relationship(
+        "Party",
+        back_populates="election",
+        uselist=True,
+        passive_deletes=True,
+        order_by="Party.name",
+    )
+
+    # Precincts in the election
+    precincts = relationship(
+        "Precinct",
+        back_populates="election",
+        uselist=True,
+        passive_deletes=True,
+        order_by="Precinct.name",
+    )
+
+    # Districts in the election
+    districts = relationship(
+        "District",
+        back_populates="election",
+        uselist=True,
+        passive_deletes=True,
+        order_by="District.name",
+    )
+
+    # Results of the election
+    results = relationship(
+        "ElectionResult",
+        back_populates="election",
+        uselist=True,
+        passive_deletes=True,
+        order_by="ElectionResult.id",
+    )
+
+    # When a user deletes an election, we keep it in the database just in case
+    # they change their mind, but flag it so that we can restrict access
+    deleted_at = Column(UTCDateTime)
+
+    @validates('polls_timezone')
+    def convert_upper(self, _key, polls_timezone):
+        return polls_timezone.upper()
+
+    __table_args__ = (UniqueConstraint("organization_id", "election_name"),)
+
+
+class ElectionResultSource(str, enum.Enum):
+    FILE = "File"
+    DATA_ENTRY = "Data Entry"
+
+class ElectionResultStatus(str, enum.Enum):
+    READY="Ready"
+    FAILED="Failed"
+    PUBLISHED="Published"
+
+class ElectionResult(BaseModel):
+    id = Column(String(200), primary_key=True)
+    total_ballots_cast = Column(String(200), nullable=False)
+    source = Column(Enum(ElectionResultSource), nullable=False)
+    status = Column(Enum(ElectionResultStatus), default=ElectionResultStatus.READY, nullable=False)
+
+    #Election to which results belong
+    election_id = Column(
+        String(200),
+        ForeignKey("election.id", ondelete="cascade"),
+        nullable=False
+    )
+    election = relationship("Election", back_populates="results")
+    #Jurisdiction to which results belong
+    jurisdiction_id = Column(
+        String(200),
+        ForeignKey("jurisdiction.id", ondelete="cascade"),
+        nullable=False
+    )
+    jurisdiction = relationship("Jurisdiction")
+    #Precinct to which results belong
+    precinct_id = Column(
+        String(200),
+        ForeignKey("precinct.id", ondelete="cascade"),
+        nullable=False
```

Comment:
are results always going to be reported by precinct?

Reply:
@benadida unsure, I just have it set up that way since that's how the UI mockup had indicated.

Reply:
@mcchilders will results always be reported by precinct? Or will there be some that are not tied to a precinct?